src/components/CreateResetPassword/CreateResetPassword.js

```diff
@@ -60,28 +71,44 @@ class CreateResetPassword extends Component {
     };
   }
 
-  componentDidMount() {
-    // Focus new-password input on mount
-    this.newPassword.current.getRenderedComponent().input.current.focus();
-  }
-
   togglePasswordMask = () => {
     this.setState(({ passwordMasked }) => ({
       passwordMasked: !passwordMasked,
     }));
   };
 
+  newPasswordFieldValidation = (valid, errorCode) => {
+    this.handleValidation(
+      valid,
+      errorCode,
+      this.translationNamespaces.smartComponents,
+    );
+  };
+
   confirmPasswordFieldValidation = (value, { newPassword, confirmPassword } = {}) => {
-    const isConfirmPasswordInvalid = newPassword && confirmPassword && newPassword !== confirmPassword;
+    const confirmPasswordValid = !(newPassword && confirmPassword && newPassword !== confirmPassword);
+
+    this.handleValidation(
+      confirmPasswordValid,
+      this.passwordMatchErrorCode,
+      this.translationNamespaces.errors,
+    );
+  };
+
+  handleValidation = (valid, errorCode, translationNamespace) => {
     const { errors } = this.props;
+    const error = {
+      code: errorCode,
+      translationNamespace,
+    };
 
-    if (!isConfirmPasswordInvalid) {
-      remove(errors, this.passwordMatchError);
+    if (valid) {
+      remove(errors, error);
```

Comment:
It is a bad practice to mutate props. The same for line 111. In this case we should have an action that triggers in particular cases and changes errors array in application state.